commands/src/main/java/org/wildfly/extras/creaper/commands/elytron/sasl/AddConfigurableSaslServerFactory.java

```diff
@@ -0,0 +1,226 @@
+            }
+            if (name.isEmpty()) {
+                throw new IllegalArgumentException("Name of the configurable-sasl-server-factory must not be empty value");
+            }
+            this.name = name;
+        }
+
+        public Builder saslServerFactory(String saslServerFactory) {
+            this.saslServerFactory = saslServerFactory;
+            return this;
+        }
+
+        public Builder protocol(String protocol) {
+            this.protocol = protocol;
+            return this;
+        }
+
+        public Builder serverName(String serverName) {
+            this.serverName = serverName;
+            return this;
+        }
+
+        public Builder addFilters(Filter... filters) {
+            if (filters == null) {
+                throw new IllegalArgumentException("Filters added to configurable-sasl-server-factory must not be null");
+            }
+            Collections.addAll(this.filters, filters);
+            return this;
+        }
+
+        public Builder addProperties(Property... properties) {
+            if (properties == null) {
+                throw new IllegalArgumentException("Properties added to configurable-sasl-server-factory must not be null");
+            }
+            Collections.addAll(this.properties, properties);
+            return this;
+        }
+
+        public Builder replaceExisting() {
+            this.replaceExisting = true;
+            return this;
+        }
+
+        public AddConfigurableSaslServerFactory build() {
+            if (saslServerFactory == null || saslServerFactory.isEmpty()) {
+                throw new IllegalArgumentException("sasl-server-factory must not be null and must include at least one entry");
+            }
+            return new AddConfigurableSaslServerFactory(this);
+        }
+
+    }
+
+    public static final class Filter {
+
+        private final String patternFilter;
+        private final String predefinedFilter;
+        private final Boolean enabling;
+
+        private Filter(FilterBuilder builder) {
+            this.patternFilter = builder.patternFilter;
+            this.predefinedFilter = builder.predefinedFilter;
+            this.enabling = builder.enabling;
+        }
+
+        public String getPatternFilter() {
+            return patternFilter;
+        }
+
+        public String getPredefinedFilter() {
+            return predefinedFilter;
+        }
+
+        public Boolean getEnabling() {
+            return enabling;
+        }
+
+    }
+
+    public static final class FilterBuilder {
+
+        private String patternFilter;
+        private String predefinedFilter;
+        private Boolean enabling;
+
+        public FilterBuilder patternFilter(String patternFilter) {
+            this.patternFilter = patternFilter;
+            return this;
+        }
+
+        public FilterBuilder predefinedFilter(String predefinedFilter) {
```

Comment:
minor note: is there a reason (I'm overlooking) why `enum` is not used instead of `String` here?